src/explore-education-statistics-admin/src/components/editable/EditableKeyStat.tsx

```diff
@@ -1,205 +1,81 @@
-                label="Guidance text"
-              />
-
-              <ButtonGroup>
-                <Button
-                  disabled={!form.isValid}
-                  type="submit"
-                  className="govuk-!-margin-right-2"
-                >
-                  Save
-                </Button>
-                <Button variant="secondary" onClick={toggleShowForm.off}>
-                  Cancel
-                </Button>
-              </ButtonGroup>
-            </Form>
-          )}
-        </Formik>
-      );
-    }
-
-    const dataDefinitionTitle = summary?.dataDefinitionTitle[0] || 'Help';
-
+  const { updateKeyStatisticDataBlock } = useReleaseContentActions();
+  if ((keyStat as KeyStatisticDataBlock).dataBlockId) {
     return (
-      <>
-        <KeyStatTile
-          title={keyStat.title}
-          value={keyStat.value}
-          testId={testId}
-          isReordering={isReordering}
-        >
-          {summary?.dataSummary[0] && (
-            <p className="govuk-body-s" data-testid={`${testId}-summary`}>
-              {summary.dataSummary[0]}
-            </p>
-          )}
-        </KeyStatTile>
-
-        {summary?.dataDefinition[0] && !isReordering && (
-          <Details
-            summary={dataDefinitionTitle}
-            className={styles.definition}
-            hiddenText={
-              dataDefinitionTitle === 'Help'
-                ? `for ${keyStat.title}`
-                : undefined
-            }
-          >
-            <div data-testid={`${testId}-definition`}>
-              {summary.dataDefinition.map(data => (
-                <ReactMarkdown key={data}>{data}</ReactMarkdown>
-              ))}
-            </div>
-          </Details>
-        )}
-
-        {isEditing && !isReordering && (
-          <ButtonGroup className="govuk-!-margin-top-2">
-            <Button onClick={toggleShowForm.on}>Edit</Button>
+      <EditableKeyStatDataBlock
+        keyStat={keyStat as KeyStatisticDataBlock}
+        testId={testId}
+        isEditing={isEditing}
+        isReordering={isReordering}
+        onRemove={onRemove}
+        onSubmit={async values => {
+          const request: KeyStatisticDataBlockUpdateRequest = {
+            trend: values.trend,
+            guidanceTitle: values.guidanceTitle,
+            guidanceText: values.guidanceText,
+          };
+          await updateKeyStatisticDataBlock({
+            releaseId: keyStat.releaseId,
+            keyStatisticId: keyStat.id,
+            request,
+          });
+        }}
+      />
+    );
+  }
 
-            {onRemove && (
-              <Button
-                disabled={removing}
-                variant="secondary"
-                onClick={() => {
-                  toggleRemoving.on();
-                  onRemove();
-                }}
-              >
-                Remove
-              </Button>
-            )}
-          </ButtonGroup>
-        )}
-      </>
+  if (
+    (keyStat as KeyStatisticText).title &&
```

Comment:
Again, nice if we could have done the cast by now so that we don't have to repeat it here.

Reply:
I'm leaving this for Nick as I'm not sure what The Way is here.

Reply:
:+1: